src/components/App/App.js

```diff
@@ -1,14 +1,95 @@
 /* eslint-disable react/jsx-filename-extension */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { ThemeProvider } from 'styled-components';
 
-import { GlobalStyles } from '../../GlobalStyles.styles';
-import HomePage from '../../pages/Home/Home.page';
+import { GlobalStyles, lightTheme, darkTheme } from '../../themes';
+import { HomePage, VideoDetailsPage } from '../../pages';
+import { gapiLoadClient } from '../../util/YouTube';
+import { items } from '../../mock/youtube-videos-mock.json';
+import { Navbar } from '../../components';
 
 function App() {
+  const [theme, setTheme] = useState(lightTheme);
+  const [darkThemeEnabled, setDarkThemeEnabled] = useState(false);
+  const [searchKeyword, setSearchKeyword] = useState('Wizeline');
+  const [YouTubeData, setYouTubeData] = useState(items);
+  const [currentPage, setCurrentPage] = useState('');
+  const [selectedVideoIndex, setSelectedVideoIndex] = useState(0);
+
+  useEffect(() => {
+    // set Google API
+    gapiLoadClient();
+  }, []);
+
+  useEffect(() => {
+    // retrieve theme from storage
+    const isDark = localStorage.getItem('darkThemeEnabled');
+    if (isDark) {
+      setDarkThemeEnabled(isDark === 'true');
+    } else {
+      localStorage.setItem('darkThemeEnabled', false);
+    }
+  }, []);
```

Comment:
This to hooks have to be separated for any reason? because they are executing only on the mounting, and they seem to not have dependency, you can use only one hook, respecting the sequence of execution:
```js
useEffect(() => {
    // set Google API
    gapiLoadClient();
    // retrieve theme from storage
    const isDark = localStorage.getItem('darkThemeEnabled');
    if (isDark) {
      setDarkThemeEnabled(isDark === 'true');
    } else {
      localStorage.setItem('darkThemeEnabled', false);
    }
}, []);
```